src/node.rs

```diff
@@ -781,6 +785,133 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
         Ok(vm_block_result)
     }
 
+    fn display_detailed_gas_info<H: HistoryMode>(
+        &self,
+        vm: &VmInstance<H>,
+        spent_on_pubdata: u32,
+    ) -> eyre::Result<()> {
+        let debug = BootloaderDebug::load_from_memory(vm)?;
+
+        println!("┌─────────────────────────┐");
+        println!("│       GAS DETAILS       │");
+        println!("└─────────────────────────┘");
+
+        // Total amount of gas (should match tx.gas_limit).
+        let total_gas_limit = debug
+            .total_gas_limit_from_user
+            .saturating_sub(debug.reserved_gas);
+
+        let intrinsic_gas = total_gas_limit - debug.gas_limit_after_intrinsic;
+        let gas_for_validation = debug.gas_limit_after_intrinsic - debug.gas_after_validation;
+
+        let gas_spent_on_compute =
+            debug.gas_spent_on_execution - debug.gas_spent_on_bytecode_preparation;
+
+        let gas_used = intrinsic_gas
+            + gas_for_validation
+            + debug.gas_spent_on_bytecode_preparation
+            + gas_spent_on_compute;
+
+        println!(
+            "Gas - Limit: {} | Used: {} | Refunded: {}",
+            to_human_size(total_gas_limit),
+            to_human_size(gas_used),
+            to_human_size(debug.refund_by_operator)
+        );
+
+        if debug.total_gas_limit_from_user != total_gas_limit {
+            println!(
+                "  WARNING: user actually provided more gas {}, but system had a lower max limit.",
+                to_human_size(debug.total_gas_limit_from_user)
```

Comment:
Since these are warnings, should we post this message in yellow font for better awareness?